tests/Authorization/AuthorizableTest.php

```diff
@@ -288,6 +288,13 @@ public function testCanCascadesToGroupsSimple(): void
         $this->assertTrue($this->user->can('admin.access'));
     }
 
+    public function testCanCascadesToGroupsMultiple(): void
```

Comment:
I want the tests to check more things.
For example, adding a few permission's with `addPermission()` and then checking the correct PR function.

Reply:
Maybe I don't understand you clearly, but why should the `addPermission()` be called in the test? Or are you trying to say that this PR may fail when a permission is added?

Reply:
@sammyskills I'm sorry if I couldn't explain my meaning clearly.

Something like this:
```php
$user->addPermission('admin.access', 'beta.access');

$res= $this->user->can('beta.access','admin.access');
$this->assertTrue($res);

$res= $this->user->can('users.edit','users.delete');
$this->assertFalse($res);

$user->removePermission('admin.access');
$res= $this->user->can('admin.access','users.delete');
$this->assertFalse($res);

$res= $this->user->can('admin.access','beta.access');
$this->assertTrue($res);
```

Reply:
@datamweb Thank you for the explanation.

`addPermission()` and `removePermission()` are already covered in the `AuthorizableTest`. Why should they be repeated here?

Reply:
My goal is not `addPermission()` or `removePermission()`, my goal is to test method `can()` if multiple threads are used.
Is this covered? If yes, link please?

Reply:
> multiple threads

What do you mean?

Reply:
> What do you mean?

`$user->can('users.create', 'users.edit', 'etc')`

Reply:
`can()` will check the user's **direct permissions (user-level permissions) first**, and **then** check against all of the user's **groups permissions** (group-level permissions) to determine if they are allowed.

Q. Did the tests cover `$user->can('users.create', 'users.edit', 'etc')` in the first case(**direct permissions (user-level permissions**) first?

Reply:
Oh, I now understand what you mean. 

The check for single permission is already covered in `testAddPermissionWithNoExistingPermissions()` and `testAddPermissionWithExistingPermissions()`, but not for multiple threads like you pointed out.

Reply:
Well, you get my point, 😂